string obj_name
bool new_msg